### PR TITLE
PR to support postgres-configuration support for sslRejectUnauthorized: false #1247

### PR DESCRIPTION
**Pull request by rakesh561**
PR to support postgres-configuration support for sslRejectUnauthorized: false #1247

##### ISSUE TYPE

 - New or Enhanced Feature

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: roles/installer/tasks/database_configuration.yml [hunk 1]
```diff
@@ -85,21 +85,22 @@
 
 - name: Set PostgreSQL Configuration
   set_fact:
```
Comment (TheRealHaoLiu) [inline roles/installer/tasks/database_configuration.yml:87]: now there's 2 set_fact

File: roles/installer/tasks/database_configuration.yml [hunk 2]
```diff
@@ -85,21 +85,22 @@
...
-    pg_config: '{{ _generated_pg_config_resources["resources"] | default([]) | length | ternary(_generated_pg_config_resources, _pg_config) }}'
```
Comment (TheRealHaoLiu) [inline roles/installer/tasks/database_configuration.yml:88]: is this removal intended?

File: roles/installer/tasks/database_configuration.yml [hunk 3]
```diff
@@ -85,21 +85,22 @@
...
-  no_log: "{{ no_log }}"
```
Comment (rooftopcellist) [inline roles/installer/tasks/database_configuration.yml:89]: I think this removal is unintended. It results in invalid syntax:
* https://github.com/ansible/awx-operator/blob/6dbe2aa134d1a4ebd03a70bc584921eff54914d6/roles/installer/tasks/database_configuration.yml#L86-L89

File: roles/installer/tasks/database_configuration.yml (hunk 2 above)
Comment (rakesh561) [reply roles/installer/tasks/database_configuration.yml:88]: That was not intended.I updated the PR set_fact got added twice in the same task.

File: roles/installer/tasks/database_configuration.yml (hunk 3 above)
Comment (rakesh561) [reply roles/installer/tasks/database_configuration.yml:89]: That was not intended.I updated the PR set_fact got added twice in the same task.

File: roles/installer/tasks/database_configuration.yml (hunk 2 above)
Comment (rakesh561) [reply roles/installer/tasks/database_configuration.yml:88]: removed the extra set_Fact that was causing issue

**Comment by rakesh561** [comment]
Any suggestions/guidance in case of issues to fix them would be much appreciated would like to contribute to the open source.

File: roles/installer/tasks/database_configuration.yml [hunk 4]
```diff
@@ -85,21 +85,21 @@
...
-
-- name: Store Database Configuration
   set_fact:
     awx_postgres_user: "{{ pg_config['resources'][0]['data']['username'] | b64decode }}"
     awx_postgres_pass: "{{ pg_config['resources'][0]['data']['password'] | b64decode }}"
     awx_postgres_database: "{{ pg_config['resources'][0]['data']['database'] | b64decode }}"
     awx_postgres_port: "{{ pg_config['resources'][0]['data']['port'] | b64decode }}"
     awx_postgres_host: "{{ pg_config['resources'][0]['data']['host'] | b64decode }}"
     awx_postgres_sslmode: "{{ pg_config['resources'][0]['data']['sslmode'] |  default('prefer'|b64encode) | b64decode }}"
+    awx_postgres_sslunauthorized: "{{ pg_config['resources'][0]['data']['sslrejectUnauthorized'] |  default('false'|b64encode) | b64decode }}"
```
Comment (fosterseth) [inline roles/installer/tasks/database_configuration.yml:102]: do you mind adding a note about this setting to the operator docs, somewhere here

https://github.com/ansible/awx-operator/blob/6513e93e7fdf117a1518392a0d838f267a67088d/README.md?plain=1#L536

you don't need to change the example spec above that line though, just a note about this param

Comment (rakesh561) [reply roles/installer/tasks/database_configuration.yml:102]: Done @fosterseth also default value i have set it to false do you think it has to be set to true and then if some one uses external database without sslrejectUnauthorized they can pass the variable false.

File: README.md [hunk 5]
```diff
@@ -533,6 +533,8 @@ type: Opaque
 
 > It is possible to set a specific username, password, port, or database, but still have the database managed by the operator. In this case, when creating the postgres-configuration secret, the `type: managed` field should be added.
 
+**Note**. The vairable `sslrejectUnauthorized1 is valid for `external` databases only. The allowed values are: `false`, `true`.
```
Comment (dsavineau) [inline README.md:536]:
```suggestion
**Note**. The variable `sslrejectUnauthorized` is valid for `external` databases only. The allowed values are: `false`, `true`.
```

**Review by dsavineau — COMMENTED** [review]
Could you share where's the `sslrejectUnauthorized` option is coming from ?

I've looked at the libpq and psycopg documentation/code and never seen such option.
The closest I found was in a postgresql nodejs binding project.

Also if this option is only needed for external database then we need to set this option only if the database type is `unmanaged`

**Comment by rakesh561** [comment]
@dsavineau seems like sslrejectUnauthorized` is not available in the python postgres library i will close this request
